org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -109,9 +110,15 @@ protected void importProjects(Collection<IPath> rootPaths, IProgressMonitor moni
 		SubMonitor subMonitor = SubMonitor.convert(monitor, rootPaths.size() * 100);
 		for (IPath rootPath : rootPaths) {
 			File rootFolder = rootPath.toFile();
-			IProjectImporter importer = getImporter(rootFolder, subMonitor.split(30));
-			if (importer != null) {
-				importer.importToWorkspace(subMonitor.split(70));
+			List<IProjectImporter> importers = new ArrayList<>(importers());
+			Iterator<IProjectImporter> iter = importers.iterator();
+			while (iter.hasNext()) {
+				IProjectImporter importer = iter.next();
+				importer.initialize(rootFolder);
+				if (importer.applies(subMonitor.split(1))) {
+					importer.importToWorkspace(subMonitor.split(70));
+				}
+				iter.remove();
```

Comment:
there's no point in removing the importer anymore. can even use a regular foreach loop from importers()

Reply:
Fixed.